sdk/tests/DeserializationTests.cc

```diff
@@ -0,0 +1,261 @@
+/*-
+ *
+ * Hedera C++ SDK
+ *
+ * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "AccountBalance.h"
+#include "AccountId.h"
+#include "ECDSAPrivateKey.h"
+#include "ECDSAPublicKey.h"
+#include "ExchangeRate.h"
+#include "ExchangeRateSet.h"
+#include "PublicKey.h"
+#include "Status.h"
+#include "TransactionReceipt.h"
+#include "TransactionRecord.h"
+#include "TransactionResponse.h"
+
+#include "impl/TimestampConverter.h"
+
+#include <gtest/gtest.h>
+#include <proto/basic_types.pb.h>
+#include <proto/exchange_rate.pb.h>
+#include <proto/transaction_receipt.pb.h>
+#include <proto/transaction_record.pb.h>
+#include <proto/transaction_response.pb.h>
+#include <proto/crypto_get_account_balance.pb.h>
+
+using namespace Hedera;
+
+class DeserializationTests : public ::testing::Test
+{
+protected:
+  [[nodiscard]] inline const uint64_t getTestShardNum() const { return mShardNum; }
+  [[nodiscard]] inline const uint64_t getTestRealmNum() const { return mRealmNum; }
+  [[nodiscard]] inline const uint64_t getTestContractNum() const { return mContractNum; }
+  [[nodiscard]] inline const int32_t  getTestCents() const { return cents; }
+  [[nodiscard]] inline const int32_t  getTestHbar() const { return hbar; }
+  [[nodiscard]] inline const uint64_t getTestSeconds() const { return seconds; }
+  [[nodiscard]] inline const Hbar     getTestBalance() const { return mBalance; }
+  [[nodiscard]] inline const AccountId& getTestAccountId() const { return mAccountId; }
+  [[nodiscard]] inline const AccountId& getTestAccountIdFrom() const { return mAccountIdFrom; }
+  [[nodiscard]] inline const AccountId& getTestAccountIdTo() const { return mAccountIdTo; }
+  [[nodiscard]] inline const std::shared_ptr<ECDSAPublicKey>& getTestECDSAPublicKey() const { return mPublicKeyFromString; }
```

Comment:
This doesn't look like it follows our formatting outlined in our `.clang-format` file. You should be able to set up your IDE to auto-format on save or on `git commit`.